Documentation updates (#3012)
openvinotoolkit/model_server: docs/genai.md

> # Endpoints for Generative Use Cases {#ovms_docs_genai}

Reviewer: This title is descriptive, but from the menu point of view it's quite long (at the first sight you don't see what it means).
Maybe we could shorten it and start with `GenAI` phrase so it catches the eye of the viewer like:
`GenAI Endpoints` or `GenAI Use Cases` ?

> OpenVINO Model Server allow extending the REST API interface to support arbitrary input format and execute arbitrary pipeline implemented as a MediaPipe graph.

Reviewer: ```suggestion
OpenVINO Model Server allows extending the REST API interface to support arbitrary input format and execute arbitrary pipeline implemented as a MediaPipe graph.
```

> The client makes a call to http://<server>:<port>/v3/... URL and post a JSON payload. The request is dispatched to the graph based on the `model` filed in the JSON content.

Reviewer 2: ```suggestion
The client makes a call to http://<server>:<port>/v3/... URL and post a JSON payload. The request is dispatched to the graph based on the `model` field in the JSON content.
```

> The client makes a call to http://<server>:<port>/v3/... URL and post a json payload. The request is dispatched to the graph based on the `model` filed in the json content.

Reviewer: ```suggestion
The client makes a call to http://<server>:<port>/v3/... URL and post a JSON payload. The request is dispatched to the graph based on the `model` filed in the JSON content.
```

This specifies OpenAI API. Should we mention it somewhere in this paragraph? Theoretically GenAI use case could work on a different API.

It supports both the unary calls with a single response per request and streams which return a series of responses per request.

> Currently, there are built-in in the model server the following use cases:

Reviewer: ```suggestion
Use cases currently built-in in the model server:
```

## Text generation

The implementation is compatible with the OpenAI API for [chat/completions](./model_server_rest_api_chat.md) and [completions](./model_server_rest_api_completions.md).

> It supports a wide range of text generation models from Hugging Faces Hub.

Reviewer: ```suggestion
It supports a wide range of text generation models from Hugging Face Hub.
```

Internally it employs continuous batching and paged attention algorithms for efficient execution both on CPU and GPU.

> Learn model about the [LLM graph configuration](./llm/reference.md) and [exporting the models from Hugging Faces for serving](../demos/common/export_models/README.md).

Reviewer: ```suggestion
Learn more about the [LLM graph configuration](./llm/reference.md) and [exporting the models from Hugging Face for serving](../demos/common/export_models/README.md).
```

Check also the demo of [text generation](../demos/continuous_batching/README.md)

## Text embeddings

> Text embeddings transforming the semantic meaning of the text into a numerical vector. This operation is crucial for text searching and algorithms like RAG (Retrieval Augmented Generation).

Reviewer: ```suggestion
Text embeddings transform the semantic meaning of the text into a numerical vector. This operation is crucial for text searching and algorithms like RAG (Retrieval Augmented Generation).
```

The model server has built-in support for requests compatible with OpenAI AI and [embeddings endpoint](./model_server_rest_api_embeddings.md).
It can run the execution both on CPU and GPU.

Check the [demo](../demos/embeddings/README.md) how this endpoint can be used in practice. 

## Documents reranking

> Reranking process is used to sort the list of documented based on relevance in the context of a query. Just like text generation and embeddings, it is essential element or RAG chains.

Reviewer: ```suggestion
Reranking process is used to sort the list of documents based on relevance in the context of a query. Just like text generation and embeddings, it is essential element or RAG chains.
```
